File: Full Source BE/src/main/java/vn/edu/hcmuaf/fit/online/shoe/shop/entity/OrderDetail.java
package vn.edu.hcmuaf.fit.online.shoe.shop.entity;

import java.sql.Timestamp;


import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Entity
@Table(name = "order_detail")
public class OrderDetail {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private long id;

	@ManyToOne
	@JoinColumn(name = "order_id")
	private Order order;

	@ManyToOne
	@JoinColumn(name = "product_id")
	@JsonIgnoreProperties(value = { "orderDetails", "handler", "hibernateLazyInitializer" }, allowSetters = true)
	private Product product;
	
	private double price;
	
	private long quantity;
	
	@CreationTimestamp
	private Timestamp createAt;
	
	@UpdateTimestamp
	private Timestamp updateAt;

	public OrderDetail() {

	}

	public OrderDetail(Order order, Product product, double price, long quantity) {
		this.order = order;
		this.product = product;
		this.price = price;
		this.quantity = quantity;
	}

	public long getId() {
		return id;
	}

	public void setId(long id) {
		this.id = id;
	}

	public Order getOrderId() {
		return order;
	}

	public void setOrderId(Order orderId) {
		this.order = orderId;
	}

	public Product getProduct() {
		return product;
	}

	public void setProduct(Product product) {
		this.product = product;
	}

	public double getPrice() {
		return price;
	}

	public void setPrice(double price) {
		this.price = price;
	}

	public long getQuantity() {
		return quantity;
	}

	public void setQuantity(long quantity) {
		this.quantity = quantity;
	}

	public Timestamp getCreateAt() {
		return createAt;
	}

	public void setCreateAt(Timestamp createAt) {
		this.createAt = createAt;
	}

	public Timestamp getUpdateAt() {
		return updateAt;
	}

	public void setUpdateAt(Timestamp updateAt) {
		this.updateAt = updateAt;
	}

	@Override
	public String toString() {
		return "OrderDetail [id=" + id + ", order=" + order + ", product=" + product + ", price=" + price
				+ ", quantity=" + quantity + ", createAt=" + createAt + ", updateAt=" + updateAt + "]";
	}

	
}

File: Full Source FE/src/app/model/orderDetail.ts
import {Product} from "./product";

export class OrderDetail{
  private _id:number;
  private _orderId:number;
  private _product:Product;
  private _price:number;
  private _quantity:number;
  private _createAt:Date;

  constructor(id: number, orderId: number, product: Product, price: number, quantity: number, createAt: Date) {
    this._id = id;
    this._orderId = orderId;
    this._product = product;
    this._price = price;
    this._quantity = quantity;
    this._createAt = createAt;
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get orderId(): number {
    return this._orderId;
  }

  set orderId(value: number) {
    this._orderId = value;
  }

  get product(): Product {
    return this._product;
  }

  set product(value: Product) {
    this._product = value;
  }

  get price(): number {
    return this._price;
  }

  set price(value: number) {
    this._price = value;
  }

  get quantity(): number {
    return this._quantity;
  }

  set quantity(value: number) {
    this._quantity = value;
  }

  get createAt(): Date {
    return this._createAt;
  }

  set createAt(value: Date) {
    this._createAt = value;
  }
}
